components/script/dom/blob.rs

```diff
@@ -70,23 +94,20 @@ impl Blob {
         Ok(Blob::new(global, bytes, &typeString.to_ascii_lowercase()))
     }
 
-    pub fn read_out_buffer(&self, send: Sender<Vec<u8>>) {
-        send.send(self.bytes.clone().unwrap_or(vec![])).unwrap();
-    }
+    pub fn get_bytes(&self) -> Vec<u8> {
+        let start = self.bytes_start.unwrap_or(0);
+        let end = self.bytes_end.unwrap_or(self.bytes.len());
 
-    // simpler to use version of read_out_buffer
-    pub fn clone_bytes(&self) -> Vec<u8> {
-        self.bytes.clone().unwrap_or(vec![])
+        self.bytes[start..end].to_vec()
```

Comment:
Hmm, I'm wrong - the library does accept `&[u8]`, so I think we should make this return `&[u8]` instead.